addon/components/power-select-infinity.ts

```diff
@@ -0,0 +1,373 @@
+     *
+     * @type {number}
+     * @argument estimateHeight
+     * @memberof PowerSelectInfinityArgs
+     */
+    estimateHeight?: number;
+    /**
+     * Label path - the path of the object to display in the text box
+     *
+     * @type {string}
+     * @memberof PowerSelectInfinityArgs
+     */
+    labelPath?: string;
+
+    /**
+     * The loading component to display
+     *
+     * @type {string}
+     * @memberof PowerSelectInfinityArgs
+     */
+    loadingComponent?: string;
+
+    /**
+     * The method invoked when `canLoadMore` is true and
+     * the bottom of the list is reached.
+     *
+     * @type {(keyword: string | null) => any[]}
+     * @argument loadMore
+     * @memberof PowerSelectInfinityArgs
+     */
+    loadMore?: (keyword?: string) => Promise<T[]>;
+    /**
+     * The message shown when no options are returned.
+     *
+     * @type {string}
+     * @argument noMatchesMessage
+     * @memberof PowerSelectInfinityArgs
+     */
+    noMatchesMessage?: string;
+
+    /**
+     * The delay for invoking the `@search` method
+     * when typing.
+     *
+     * @type {number}
+     * @argument searchDebounceDelay
+     * @memberof PowerSelectInfinityArgs
+     */
+    searchDebounceDelay?: number;
+    /**
+     * Used by ember-vertical-collection for occlusion rendering.
+     *
+     * @type {boolean}
+     * @argument staticHeight
+     * @memberof PowerSelectInfinityArgs
+     */
+    staticHeight?: boolean;
+}
+
+export default class PowerSelectInfinity<T> extends Component<PowerSelectInfinityArgs<T>> {
+    get allowClear() {
+        return this.args.allowClear ?? true;
+    }
+
+    get bufferSize() {
+        return this.args.bufferSize ?? 5;
+    }
+
+    get canLoadMore() {
+        return this.args.canLoadMore ?? true;
+    }
+
+    get estimateHeight() {
+        return this.args.estimateHeight ?? 30;
+    }
+
+    get loadingComponent() {
+        return this.args.loadingComponent ?? 'power-select-infinity/loading';
+    }
+
+    get matchTriggerWidth() {
+        return this.args.matchTriggerWidth ?? true;
+    }
+
+    get noMatchesMessage() {
+        return this.args.noMatchesMessage ?? 'No results found';
+    }
+
+    get optionsComponent() {
+        return this.args.optionsComponent ?? 'power-select-infinity/options';
+    }
+
+    get search() {
+        return this.args.search ?? false;
+    }
+
+    get searchDebounceDelay() {
+        return this.args.searchDebounceDelay ?? 300;
+    }
+
+    get searchEnabled() {
+        return this.args.searchEnabled ?? true;
+    }
+
+    get searchField() {
+        return this.args.searchField ?? undefined;
+    }
+
+    get staticHeight() {
+        return this.args.staticHeight ?? false;
+    }
+
+    get tabindex() {
+        return this.args.tabindex ?? -1;
+    }
+
+    @tracked isLoadingMore = false;
+    @tracked loading = false;
+    @tracked searchText = '';
+    @tracked select: Select | null = null;
+
+    get showCreateMessage() {
```

Comment:
I think there's an argument to be made here that `showCreateMessage` could still be useful even if records are returned.

For example, in CC there's a global patient search with "create" functionality. When you search someone, say "Adam Baker", there could be results; but not the one you need (different SSN, date of birth, etc)